src/App.js

```diff
@@ -36,7 +39,9 @@ class App {
 		}
 
 		// Attach popup
-		$( 'body' ).append( this.revisionPopup.$element );
+		$( 'html' ).addClass( 'wwt-popup' );
+		$( 'body' ).append( $overlay );
+		$overlay.append( this.revisionPopup.$element );
```

Comment:
It does seem strange that OO.ui.mixin.PopupElement doesn't handle the overlay. Anyway, given that it doesn't, should we fix that upstream, or just include something like this in the RevisionPopupWidget constructor?:

```js
this.$overlay = ( config.$overlay === true ? OO.ui.getDefaultOverlay() : config.$overlay ) || this.$element;
this.$overlay.append( this.popup.$element );
```

Reply:
I am not sure if there's an issue adding this upstream; it might be that it's expected that if you have an overlay, you (the implementer) add it yourself, and that the ButtonPopupWidget does it for you as the "implementer of the popup".

But we should raise this and see if this can be upstreamed.

I like your idea of doing this in the config, but I think that if we append to the overlay consistently it's an okay solution for now anyways.